server/jetstream_cluster.go

```diff
@@ -3466,6 +3466,22 @@ func (s *Server) jsClusteredStreamUpdateRequest(ci *ClientInfo, acc *Account, su
 		return
 	}
 
+	// Check for subject collisions here.
+	for _, sa := range cc.streams[acc.Name] {
+		if sa == osa {
```

Comment:
This same loop (without this check) exists in jsClusteredStreamRequest(). Could have been made a little function, although this check is specific to jsClusteredStreamUpdateRequest() I presume.

Reply:
Agree, we could make a function but this one does need to skip the original assignment unlike create.